src/main/haskell/ast/src/Data/Kore/AST.hs

```diff
@@ -0,0 +1,563 @@
+{-|
+Module      : Data.Kore.AST
+Description : Data Structures for representing the Kore language AST
+Copyright   : (c) Runtime Verification, 2018
+License     : UIUC/NCSA
+Maintainer  : traian.serbanuta@runtimeverification.com
+Stability   : experimental
+Portability : POSIX
+
+This module includes all the data structures necessary for representing
+all the syntactic categories of a Kore definition.
+
+Please refer to Section 9 (The Kore Language) of the
+<http://github.com/kframework/kore/blob/master/docs/semantics-of-k.pdf Semantics of K>.
+
+
+-}
+module Data.Kore.AST where
+
+import Data.Typeable
+
+data MetaType
```

Comment:
It seems that here we use the name `MetaType` to refer to the "object-level" or "meta-level" in the Semantics of K document. I'm OK with it but it looks a bit confusing. Probably using something like "Level"?